docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplate.java

```diff
@@ -227,10 +232,14 @@ public DockerSlave provision(StreamTaskListener listener) throws IOException, De
 
         logger.println("Launching " + image );
 
-        int numExecutors = 1;
         Node.Mode mode = Node.Mode.NORMAL;
 
-        RetentionStrategy retentionStrategy = new OnceRetentionStrategy(idleTerminationMinutes());
+        RetentionStrategy retentionStrategy = null;
+        if (numExecutors == 1) {
+            retentionStrategy = new OnceRetentionStrategy(idleTerminationMinutes());
+        } else {
+            retentionStrategy = new CloudRetentionStrategy(idleTerminationMinutes());
+        }
```

Comment:
I'm not 100% sure if this answers your question, but hopefully this will make sense. I have not tested with OnceRetentionStrategy and multiple executors. From the [source](https://github.com/jenkinsci/durable-task-plugin/blob/master/src/main/java/org/jenkinsci/plugins/durabletask/executors/OnceRetentionStrategy.java) and docs, it looked like it very much expected to be running on a slave with one executor. As far as I could tell from the source, as soon as *any* task was completed, it would try and terminate the slave. What I didn't want to happen was the following scenario:

* DockerTemplate configured with two executors and OnceRetentionStrategy
* Jenkins queues two jobs, QuickJob and LongJob
* Jenkins spins up a new slave with that template and assigns QuickJob to one executor and LongJob to the other on that slave.
* QuickJob completes quickly and triggers a terminate long before LongJob completes.

I wasn't sure what would happen at this point, whether or not it would let LongJob finish up before tearing down or if it would just get killed. If it's the latter, then we certainly can't allow Once with multiple executors, if it's the former, then it'd be fine (but it doesn't seem significantly different than running as Cloud with a low idle termination time).

As for the case where we're running a ContinuableExecutable and a Cloud strategy, I don't have that in my system and am not familiar enough with the paradigm to speak to it, so it is untested. (Do such tasks assume/require that they be built on the same node?)